src/screens/PinFlow/CurrentPinScreen.tsx

```diff
@@ -14,37 +24,88 @@ interface Props {
   appSettings: {
     isPinSet: boolean;
   };
+  setTimeCounter: (timestamp: number) => SetTimeCounterAction;
+  setFailedAttempts: (attempt: number) => SetFailedAttemptsAction;
+  timeCounter: TimeCounterState;
 }
 
 interface State {
   pin: string;
   error: string;
+  failedTimes: number;
+  isCount: boolean;
 }
 
-export class CurrentPinScreen extends PureComponent<Props, State> {
+class CurrentPinScreen extends PureComponent<Props, State> {
   state = {
     pin: '',
     error: '',
+    failedTimes: 0,
+    isCount: true,
+  };
+
+  handleFailedAttempt = (increasedFailedTimes: number) => {
+    if (increasedFailedTimes > 2) {
+      const { attempt } = this.props.timeCounter;
+      const timestamp = new Date().getTime();
+      let failedAttemptTimestamp = 60 * 1000;
+      if (attempt === 1) {
+        failedAttemptTimestamp = failedAttemptTimestamp * 2;
+      } else if (attempt > 1) {
+        failedAttemptTimestamp = failedAttemptTimestamp * 10;
+      }
+      failedAttemptTimestamp = failedAttemptTimestamp + timestamp;
```

Comment:
we have dayjs library for managing time, we shouldn't try to this manually as it can get buggy